frontend/pages/TagView.js

```diff
@@ -1,14 +1,181 @@
+                    mapSquareBucket = "map-square box-four";
+                } else if (numberOfPhotos <= maxNumOfPhotos) {
+                    mapSquareBucket = "map-square box-five";
+                }
+
+                return (
+                    <Rectangle
+                        className={numberOfPhotos === 0 ? "map-grid" : mapSquareBucket}
+                        key={index}
+                        bounds={mapSquareBounds}>
+                        {// Shows map square numbers on each map square
+                            // <Marker position={L.polygon(mapSquareBounds).getBounds().getCenter()}
+                            //icon={L.divIcon({html: index})} />
+                        }
+                        <Popup>
+                            Map Square {index} <br/>
+                            <a href={link}>{numberOfPhotos} photos to show</a>
+                        </Popup>
+                    </Rectangle>
+                );
+            })
+        }
+    </>);
+}
+
+function arrondissementsOverlay(data) {
+    return data !== null ? data.map(tract => {
+        return (
+            <GeoJSON
+                style={{
+                    fillColor: "none",
+                    color: "#20CCD7"
+                }}
+                key={tract.properties["GISJOIN"]}
+                data={tract}
+            />
+        );
+    }) : <></>;
+}
 
 class TagView extends PhotoViewer {
     constructor(props) {
         super(props);
+
+        this.state = {
+            mapData: null,
+            geojsonData: null,
+            filledMapSquares: null,
+            filledMapSquaresData: null,
+            isLgViewportUp: null,
+            mapSquare: null,
+            mapLat: 48.858859,
+            mapLng: 2.3470599,
+            photos: JSON.parse(this.props.tagPhotos),
+            photographers: []
+        };
     }
 
+    async componentDidMount() {
+        try {
+
+            const mapResponse = await fetch("/api/all_map_squares/");
+            const mapData = await mapResponse.json();
+
+            for (const mapSquare of mapData) {
+                // This code right here might cause problems if said user hasn't run syncdb
+                const roughCoords = mapSquare.coordinates;
+
+                // If the map square has coordinates in the spreadsheet,
+                // it pulls those coordinates and makes those the coordinates of the marker
+                // Coords must be in (lat, lng)
+
+                // If the map square does not have the coordinates it sets them to (0, 0)
+                // NOTE(ra): this no longer happens, so we can probably remove this safety check
+                if (roughCoords) {
+                    const roughCoordsList = roughCoords.split(", ");
+                    const lat = parseFloat(roughCoordsList[0]);
+                    const lng = parseFloat(roughCoordsList[1]);
+                    mapSquare.topLeftCoords = {
```

Comment:
Nitpick but no need to repeat this block in both the `if` and `else` conditions: 

```
mapSquare.topLeftCoords = {
     lat,
     lng
};
```